discovery/sync_manager.go

```diff
@@ -755,16 +755,22 @@ func (m *SyncManager) GossipSyncers() map[route.Vertex]*GossipSyncer {
 
 // gossipSyncers returns all of the currently initialized gossip syncers.
 func (m *SyncManager) gossipSyncers() map[route.Vertex]*GossipSyncer {
-	numSyncers := len(m.inactiveSyncers) + len(m.activeSyncers)
+	numSyncers := len(m.inactiveSyncers) + len(m.activeSyncers) +
+		len(m.pinnedActiveSyncers)
 	syncers := make(map[route.Vertex]*GossipSyncer, numSyncers)
 
 	for _, syncer := range m.inactiveSyncers {
 		syncers[syncer.cfg.peerPub] = syncer
 	}
+
 	for _, syncer := range m.activeSyncers {
 		syncers[syncer.cfg.peerPub] = syncer
 	}
 
+	for _, syncer := range m.pinnedActiveSyncers {
```

Comment:
I think this is a good bug fix, and should go in either version of the fix we end up going with.

Reply:
yeah I think it will bypass filters and get all the updates. This fix seems to dedup the messages sent a bit.